docs/topics/api/auth_internal.rst

```diff
@@ -0,0 +1,62 @@
+.. _api-auth-internal:
+
+=========================
+Authentication (internal)
+=========================
+
+This documents how to use authentication in your API requests when you are
+working on a web application that lives on AMO domain or subdomain. If you
+are looking for how to authenticate with the API from an external client, using
+your API keys, read the :ref:`documentation for external authentication
+<api-auth>` instead.
+
+When using this authentication mechanism, the server is responsible for
+creating a `JSON Web Token (JWT)`_ when the user logs in, and sends it back in
+the response. The clients must then include that token as an ``Authorization``
+header on requests that need authentication. The clients never generate JWTs
+themselves.
+
+Fetching the JWT
+================
+
+A fresh JWT, valid for 30 days, is automatically generated and added to the
+responses of the following endpoints:
+
+    * ``/api/v3/accounts/login/``
+    * ``/api/v3/accounts/register/``
+    * ``/api/v3/accounts/authenticate/``
+
+The token is available in two forms:
+
+    * For the endpoints returning JSON, as a property called ``token``.
+    * For all endpoints, as a cookie called ``jwt_api_auth_token``. This cookie
+      expires when the browser is closed and is set as ``HttpOnly``.
```

Comment:
If we started using it for the frontend then we'd need to start prompting for login on browser close again since they'd be able to access authenticated pages through django but not through AJAX.